test/Tester/MethodInterceptionTests.cs

```diff
@@ -25,5 +29,38 @@ public class MethodInterceptionTests : HostedTestClusterEnsureDefaultStarted
             result = await grain.SayHello();
             Assert.Equal("Hello", result);
         }
+
+        [Fact]
+        public async Task GenericGrainMethodInterceptionTest()
+        {
+            var grain = GrainFactory.GetGrain<IGenericMethodInterceptionGrain<int>>(
+                0,
+                typeof(GenericMethodInterceptionGrain<>).Namespace + "." + nameof(GenericMethodInterceptionGrain<int>));
```

Comment:
It's a subtlety in how grain types are resolved, but I fixed it by having the tricky grain class not inherit from the generic grain class. If we decide it's a priority, I may go and fix some deficiencies in how type resolution works.